scripts/leadtime.ts

```diff
@@ -118,6 +81,50 @@ const script: Script = async (config: Config, argv: Argv) => {
     } else {
         console.error("Tickets were not fetched properly :/");
     }
-};
+}
+
+namespace script {
+    function prettyPrintDate(date: Date): string {
+        return `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
+    }
+
+    function prettyPrintTimes(values: { [key: string]: number }, statuses: string[]): string {
+        return statuses
+            .map(s => s.toLowerCase())
+            .map(s => values[s] || 0)
+            .join(",");
+    }
+
+    export function getIssueTimeStrings<IssueWithChangelog extends Issue & HasChangelog>(
+        issues: IssueWithChangelog[],
+        statuses: string[],
+        finalStatuses: string[],
+        showSummary: boolean
+    ): string[] {
+        const headings = [];
```

Comment:
I would separate this pushing and joining logic to some util method. Now it takes quite many rows just to build a header row. And of course the same goes for building the actual lines. Push and join trick is in different abstraction level than rest of the code in the function so it's hard to see the forest from the trees.